### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by scotthasbrouck**

### CODE REVIEW

**Comment by scotthasbrouck** [comment]
@dmitrig01 Forgot to commit and push this up the other day. Note that this is not an integration test on the minified build. The complication is that there is no way to stub out the jsonp request, unless we were to not minify the method names of _server(), and createScript(). XHR would work fine with the minified build because of Sinon's FakeXHR thing. I'll also keep brainstorming to see if I can come up with other options.

**Comment by dmitrig01** [comment]
awesome! i'm pretty backed up but i may have time to review tomorrow

**Comment by scotthasbrouck** [comment]
Cool no rush

**Comment by dmitrig01** [comment]
hey - yknow what we shoudl do?

in all the autogenerated files which don't look like auto-genertaed files (e.g. README, this test file, the testbed/example.html), we should put a `<!-- THIS FILE IS AUTOMATICALLY GENERATED. DO NOT EDIT; EDIT xxxx INSTEAD -->`. I can already see the mistakes that will come from this :P

File: test/4_integration.js [hunk 1]
```diff
@@ -1,108 +1,242 @@
-goog.require('utils');
-goog.require('Server');
-goog.require('Queue');
-goog.require('Branch');
-goog.require('resources');
 goog.require('config');
-goog.require('storage');
-
+goog.require('utils');
```
Comment (dmitrig01) [inline test/4_integration.js]: do we need config and utils?

File: test/integration-test.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+</head>
+<body>
+	<div id="mocha"></div>
+	<link rel="stylesheet" type="text/css" href="../node_modules/mocha/mocha.css" />
+	<script src="../compiler/library/closure-library-master/closure/goog/base.js"></script>
+	<script>
+		goog.addDependency("../../../../../src/0_config.js", ['config'], []);
+		goog.addDependency("../../../../../src/0_utils.js", ['utils'], []);
+		goog.addDependency("../../../../../src/0_storage.js", ['storage'], []);
+	</script>
```
Comment (dmitrig01) [inline test/integration-test.html]: same question here - do we need the closure library or these dependencies?

**Comment by dmitrig01** [comment]
also here - let's add to changelog

File: test/4_integration.js [hunk 3]
```diff
@@ -1,108 +1,242 @@
...
-			branch.init(branch_sample_key);
-			requests[0].callback(null, browser_fingerprint_id);
-			requests[1].callback(null, { session_id: session_id, browser_fingerprint_id: browser_fingerprint_id, identity_id: identity_id });
-			requests = [];
-		}
+		it('should store in session and call open with link_identifier from hash', function(done) {
+			var assert = testUtils.plan(1, done);
+			testUtils.go("#r:12345");
+			branch.init(branch_sample_key, function(err, data) {
+				assert.equal(utils.readStore(branch._storage).click_id, '12345', 'click_id from link_identifier hash stored in session_id');
```
Comment (scotthasbrouck) [inline test/4_integration.js]: Yep, I need utils here to test that the hash value was stored

Comment (dmitrig01) [inline test/4_integration.js]: ahh ok cool. hmm... it's like a weird coupling of unit and integration testing. let's find other ways to check this -- for example, do a sendSMS and check the url it gets sent to

**Comment by scotthasbrouck** [comment]
@dmitrig01:
- Added changelog note
- removed utils and session dependency, and just checking URLs everywhere. Note that I'm still using config, only to grab the current SDK version to check the URL for _r
- added the README notice

**Comment by dmitrig01** [comment]
Hmm, I resolved the merge conflicts, but the tests don't pass for me

**Comment by dmitrig01** [comment]
can you take a look?

**Comment by scotthasbrouck** [comment]
all set!

**Comment by dmitrig01** [comment]
sweet!

**Pull request merged**